lib/web_ui/lib/src/engine/profiler.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.6
+part of engine;
+
+typedef OnBenchmark = void Function(String name, num value);
+
+/// The purpose of this class is to facilitate communication of
+/// profiling/benchmark data to the outside world (e.g. a macrobenchmark that's
+/// running a flutter app).
+///
+/// To use the [Profiler]:
+///
+/// 1. Enable the `FLUTTER_WEB_ENABLE_PROFILING` flag.
```

Comment:
nit: technically it's not a flag, but an environment variable that can be set to "true".